### PR TITLE
Add integral_error for models.spectral

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR adds the `integral_error` function into ~gammapy.modeling.models.spectral, which will calculate the error of the integral flux for a given spectral model in a given energy range.
A test for a `PowerLawSpectralModel` has also been added.

### CODE REVIEW

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @fabiopintore, one important comment already: 
`.integral_error()` hast work with arrays of emin and emax, please take a look a the prototype implementation again and adapt accordingly.

File: gammapy/modeling/models/spectral.py [hunk 1]
```diff
@@ -163,6 +163,26 @@ def integral(self, emin, emax, **kwargs):
...
+        emin, emax : `~numpy.array`
+            Arrays of minimum and maximum energies.
+
+        Returns
+        -------
+        flux, flux_err : tuple of `~astropy.units.Quantity`
+            Integral flux and flux error betwen emin and emax.
+        """
+        dE = emax - emin
+        flux = np.sum(self.integral(emin,emax))
```
Comment (adonath) [inline gammapy/modeling/models/spectral.py:181]: The idea is, that `flux` as well as `flux_err` can be arrays as well. You can either achieve this by using relative error propagation like `df = f * dg ./ df` or using the approach outlined in the "Flux" section here: https://github.com/gammapy/gammapy-extra/blob/master/experiments/uncertainty_estimation_prototype.ipynb

File: gammapy/modeling/models/spectral.py [hunk 2]
```diff
@@ -163,6 +163,24 @@ def integral(self, emin, emax, **kwargs):
...
+        emin, emax : `~numpy.array`
+            Arrays of minimum and maximum energies.
+
+        Returns
+        -------
+        flux, flux_err : tuple of `~astropy.units.Quantity`
+            Integral flux and flux error betwen emin and emax.
+        """
+        flux = self.integral(emin,emax)
+        flux_err = flux * self.evaluate_error(emin, epsilon=1e-4)[1] / self(emin)
```
Comment (adonath) [inline gammapy/modeling/models/spectral.py:181]: Minor: as `evaluate_error()` returns both dude as well as dude_err, there is no need to recompute it. You can just use  `dnde, dnde_err = self.evaluate_error()` and maybe use the center energy like `np.sqrt(e_min * e_max)`

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @fabiopintore, looks mostly good to me now. Can you maybe add one more test case. E.g. using an exponential cutoff model?

File: gammapy/modeling/models/spectral.py (hunk 2 above)
Comment (fabiopintore) [reply gammapy/modeling/models/spectral.py]: Done.

**Review by adonath — APPROVED** [review]
Thanks a lot @fabiopintore, no further comments from my side!

**Pull request merged**